code/game/objects/effects/enginespawner.dm

```diff
@@ -0,0 +1,50 @@
+//random engine spawner. takes random engines from their appropriate map file and places them. the engine will spawn with the spawner in the bottom left corner
+
+/obj/effect/spawner/engine
```

Comment:
...why? why all this copypasted code? when a normal random room spawner would work just fine?

Reply:
For better organization? I thought that if i just used the normal maint code i would get labelled as lazy, so i thought it would be better to create the proper engine spawner.